marklogic-data-hub-central/ui/src/components/load/load-list.test.tsx

```diff
@@ -109,4 +119,152 @@ describe('Load data component', () => {
+            canWriteFlow={true}
+            addStepToFlow={jest.fn()}
+            addStepToNew={jest.fn()} />
+        </AuthoritiesContext.Provider>
+      </MemoryRouter>
+    )
+
+    //Check if the list is rendered properly
+    expect(getByText('testLoadXML')).toBeInTheDocument();
+
+    fireEvent.mouseOver(getByLabelText('testLoadXML-add-icon')); // Hover over the Add to Flow Icon to get more options
+
+    //Verify if the flow related options are availble on mouseOver
+    await waitForElement(() => expect(getByTestId(`testLoadXML-toNewFlow`))); // check if option 'Add to a new Flow' is visible
+    await waitForElement(() => expect(getByTestId(`testLoadXML-toExistingFlow`))); // check if option 'Add to an existing Flow' is visible
+
+    //Click on the select field to open the list of existing flows.
+    fireEvent.click(getByTestId('testLoadXML-toNewFlow')); 
+
+    //Check if the /tiles/run/add route has been called
+    wait(() => {
+      expect(mockHistoryPush).toHaveBeenCalledWith('/tiles/run/add');
+    })
+    //TODO- E2E test to check if the Run tile is loaded or not.
+
+  })
+
+  test('Verify Load list allows step to be added to flow with writeFlow authority', async () => {
+    const authorityService = new AuthoritiesService();
+    authorityService.setAuthorities(['readIngestion','writeFlow']);
+    const mockAddStepToFlow = jest.fn();
+    const mockAddStepToNew = jest.fn();
+    const mockCreateLoadArtifact = jest.fn();
+    const mockDeleteLoadArtifact = jest.fn();
+    const {getByText, getByLabelText, getByTestId} = render(<MemoryRouter><AuthoritiesContext.Provider value={authorityService}><LoadList
+      addStepToFlow={mockAddStepToFlow}
+      addStepToNew={mockAddStepToNew}
+      canReadOnly={authorityService.canReadLoad()}
+      canReadWrite={authorityService.canWriteLoad()}
+      canWriteFlow={authorityService.canWriteFlow()}
+      createLoadArtifact={mockCreateLoadArtifact}
+      data={data.loadData.data}
+      deleteLoadArtifact={mockDeleteLoadArtifact}
+      flows={data.flows}/>
+    </AuthoritiesContext.Provider></MemoryRouter>);
+
+    const loadStepName = data.loadData.data[0].name;
+
+    fireEvent.mouseOver(getByLabelText('testLoad-add-icon'));
+
+    //verify components and text appear on hover
+    await waitForElement(() => expect(getByTestId(`${loadStepName}-toNewFlow`)));
+    await waitForElement(() => expect(getByTestId(`${loadStepName}-toExistingFlow`)));
+    await waitForElement(() => expect(getByTestId(`${loadStepName}-flowsList`)));
+    await waitForElement(() => expect(getByText('Add step to a new flow')));
+    await waitForElement(() => expect(getByText('Add step to an existing flow')));
+
+    // test adding to existing flow
+    fireEvent.click(getByTestId(`${loadStepName}-flowsList`));
+    fireEvent.click(getByText(data.flows[0].name));
+    fireEvent.click(getByText('Yes'));
+    expect(mockAddStepToFlow).toBeCalledTimes(1);
+
+    //TODO: Mock addStepToNew not implemented yet
```

Comment:
Is the "add to new" testable now? Looks like we do something similar already in load-card.test.

Reply:
@wooldridge Right, was referring to the load-card.test where it is also still a 'TODO'. Seems like while the POST for addStepToFlow is supported and testable, the POST for addStepToNew is not implemented yet for us to test. Instead, adding a step to a new flow is done with a Link to /tiles/run/add. Not 100% sure but I think it makes sense, can refer to here as well. 

https://github.com/marklogic/marklogic-data-hub/blob/aea4246752d809fdce709317d841a249509a7d8f/marklogic-data-hub-central/ui/src/pages/Load.tsx#L112

Reply:
Thanks!